server/src/com/thoughtworks/go/server/GoServer.java

```diff
@@ -76,7 +76,9 @@ protected void startServer() throws Exception {
 
     AppServer configureServer() throws Exception {
         Constructor<?> constructor = Class.forName(systemEnvironment.get(SystemEnvironment.APP_SERVER)).getConstructor(SystemEnvironment.class, String.class, SSLSocketFactory.class);
-        AppServer server = ((AppServer) constructor.newInstance(systemEnvironment, password, sslSocketFactory));
+        String value = System.getenv("KEYSTORE_PASSWORD");
+        String pass = (value != null)?value:password;
+        AppServer server = ((AppServer) constructor.newInstance(systemEnvironment, pass, sslSocketFactory));
```

Comment:
Also once in SystemEnvironment you can use

https://github.com/gocd/gocd/blob/e7c3a0ceb8d00eecd99c1927fb8a0a23608583e5/base/src/com/thoughtworks/go/util/SystemEnvironment.java#L482